CONTRIBUTING.md

```diff
@@ -1,12 +1,80 @@
+
+**During pull request review**
+A core contributor will review your pull request and provide feedback. To ensure that there is not a large backlog of inactive PRs, the pull request will be marked as stale after two weeks of no activity. After another four days, it will be closed.
+
+### Resources to help you get started
+Here are some resources to help you get started on how to contribute code or new content.
+* Look at the [Contributor documentation](/docs/README.md) to get started on building the source code on your own.
+* ["Help wanted" issues](https://github.com/dotnet/upgrade-assistant/labels/help%20wanted) - these issues are up for grabs. Comment on an issue if you want to create a fix.
+* ["Good first issue" issues](https://github.com/dotnet/upgrade-assistant/labels/good%20first%20issue) - we think these are a good for newcomers.
+
+### Identifying the scale
+
+If you would like to contribute to upgrade-assistant, first identify the scale of what you would like to contribute. If it is small (grammar/spelling or a bug fix) feel free to start working on a fix. If you are submitting a feature or substantial code contribution, please discuss it with the team and ensure it follows the product roadmap. You might also read these two blogs posts on contributing code: [Open Source Contribution Etiquette](http://tirania.org/blog/archive/2010/Dec-31.html) by Miguel de Icaza and [Don't "Push" Your Pull Requests](https://www.igvita.com/2011/12/19/dont-push-your-pull-requests/) by Ilya Grigorik. All code submissions will be rigorously reviewed and tested further by the upgrade-assistant team, and only those that meet an extremely high bar for both quality and design/roadmap appropriateness will be merged into the source.
+
+### Submitting a pull request
+
+You will need to sign a [Contributor License Agreement](https://cla.dotnetfoundation.org/) when submitting your pull request. To complete the Contributor License Agreement (CLA), you will need to follow the instructions provided by the CLA bot when you send the pull request. This needs to only be done once for any .NET Foundation OSS project.
+
+If you don't know what a pull request is read this article: https://help.github.com/articles/using-pull-requests. Make sure the repository can build and all tests pass. Familiarize yourself with the project workflow and our coding conventions. The coding, style, and general engineering guidelines are published on the [Engineering guidelines](https://github.com/dotnet/aspnetcore/wiki/Engineering-guidelines) page.
+
+### Tests
+
+-  Tests need to be provided for every bug/feature that is completed.
+-  Tests only need to be present for issues that need to be verified by QA (for example, not tasks)
+-  If there is a scenario that is far too hard to test there does not need to be a test for it.
```

Comment:
This (and the following bullet) is another area that seems gray to me.  On the one hand, I think this bullet is true. But on the other hand, I think cases that are both well-designed and too hard to test are rare enough that this bullet isn't necessary.

I think most of upgrade-assistant should be unit testable. I think enough of it should be unit testable that it might be better to just have as-needed conversations in PRs that are difficult to test for some reason. Often, "far too hard to test" means "spaghetti code" or "not designed with testing in mind," both of which can be fixed without needing a "it's too hard to test" excuse for not having unit tests.

Reply:
Removed this line